api/rest/client/methods.go

```diff
@@ -463,18 +484,35 @@ func (sf *statusFilter) pollStatus(ctx context.Context, c Client, fp StatusFilte
 	}
 }
 
-func statusReached(target api.TrackerStatus, gblPinInfo *api.GlobalPinInfo) (bool, error) {
+func statusReached(target api.TrackerStatus, gblPinInfo *api.GlobalPinInfo, rplMin int) (bool, error) {
+	// Specific case: return error if there are errors
 	for _, pinInfo := range gblPinInfo.PeerMap {
 		switch pinInfo.Status {
-		case target:
-			continue
-		case api.TrackerStatusUndefined, api.TrackerStatusClusterError, api.TrackerStatusPinError, api.TrackerStatusUnpinError:
+		case api.TrackerStatusUndefined,
+			api.TrackerStatusClusterError,
+			api.TrackerStatusPinError,
+			api.TrackerStatusUnpinError:
 			return false, fmt.Errorf("error has occurred while attempting to reach status: %s", target.String())
-		case api.TrackerStatusRemote:
-			if target == api.TrackerStatusPinned {
-				continue // to next pinInfo
+		}
+	}
+
+	// Specific case: waiting to pin. Wait until minimum replication
+	// factor reached.
+	if target == api.TrackerStatusPinned {
+		pinned := 0
+		for _, pinInfo := range gblPinInfo.PeerMap {
+			if pinInfo.Status == api.TrackerStatusPinned {
+				pinned++
 			}
-			return false, nil
+		}
+		return pinned >= rplMin, nil
+	}
```

Comment:
In the case of local, rplMin = 1. This will mean blocking until the local peer reports pinned.